fix: Load control & dialog assets with `cart.js`

surajshetty3416 opened the pull request:
Previously, control & dialog assets were getting loaded only on visiting "/cart", but since cart.js sets up various events on the homepage, dialog-related assets are required up-front.

fixes https://github.com/frappe/frappe/issues/15869
![image](https://user-images.githubusercontent.com/13928957/152364142-25f5d37d-10c9-44bd-9787-21025ea0622a.png)




---

The issue was introduced via https://github.com/frappe/erpnext/pull/27923


----------------------------------
## Final Fix:
**Issue:**
- `cart.js` was added to the erpnext-web bundle, so it loaded on every single page
- It requires controls bundle from frappe to make dialogs in it's file, which it relies on `cart.html` for
- Since only js was loaded on login, the lack of controls bundle broke the  js asset > and thus the login page wouldn't work (static page)
- This file shouldn't have been in the bundle since it is specific to `cart.html`, not a generic utility file unlike the others.

**Fix:**
- Move `cart.js` to /templates/pages with `cart.html`. They are tightly bound and exclusive to eachother
- Remove cart.js import from bundle

surajshetty3416 committed:
fix: Load control & dialog assets with cart.js

Previously, control & dialog assets were getting loaded only on visiting "/cart", but since cart.js sets up various events on homepage, dialog related assets are required up-front.

marination commented:
@surajshetty3416 I think cart.js should not be in the bundle, I must have absent mindedly put that in. Its js for `cart.html`. Not generic enough to be in the bundle. 
Relocating the file and removing the import in the bundle would solve this issue.
It should not impact the login page 🥲 
I'll relocate it and test if it breaks anything

marination committed:
fix: Remove `cart.js` from erpnext-web bundle

- Due to it being in the bundle, controls bundle was required for Dialogs used in cart
- But this is exclusively used in `cart.html`
- It has no reason to load and break the login page
- Moved it back to where the html is, it is now back to being tightly coupled with its html

marination commented:
Login works:
![2022-02-03 21 44 37](https://user-images.githubusercontent.com/25857446/152382365-8db5137a-df8b-469b-b66a-1a3655e6cef7.gif)

Cart works too:
![2022-02-03 21 45 44](https://user-images.githubusercontent.com/25857446/152382578-bc3b3173-c198-4bd3-820a-07d9037f35eb.gif)
 
Can confirm for other pages too (Source: trust me)

marination committed:
Merge branch 'develop' into fix-web-portal

marination commented:
@Mergifyio backport version-14-beta-pre-release
